### DISCUSSION TITLE
Abysmal performance

### DISCUSSION THREAD

**Discussion by victorkamoto** [abenz1267/walker#127] [General]
I really like this launcher, and I commend @abenz1267 for the hardwork and time he's put into it. The performance however is really concerning. A simple app search takes forever even with `exec-once=walker --gapplication-service`. The CPU spikes during this period,  and results are shown eventually, but the time it takes makes it unusable and I've regrettably been forced to have an alternative launcher. Is anyone else experiencing this? or am I missing something?

**Comment by abenz1267** [comment]
Can you post more about your system? Specs, distro... and how many desktop applications you have installed?

https://github.com/user-attachments/assets/9ff4c36b-8e9f-45b2-9f7b-1f14a73d6513

This is recorded on my system... Arch, 7950x, 32gb memory. 82 desktop files in /usr/share/applications.

**Reply by victorkamoto** [reply]
Wow, it looks quite snappy on your system!

`find . -type f | wc -l`
I have roughly 215 desktop files in /usr/share/applications, and 68 desktop files in ~/.local/share/applications

System: Arch, Hyprland, i7-8550U, 16gb memory, HDD

Here's a recording of my screen. Excuse the quality I had to do alot of compression.

https://github.com/user-attachments/assets/e8e04597-41b2-4a8a-ba91-3e0b3d8a2b91

**Reply by abenz1267** [reply]
Could you upload all the files somewhere? So i can test on my system and see where the problem actually is.

**Reply by victorkamoto** [reply]
Sure, [desktop-files.zip](https://github.com/user-attachments/files/17057210/desktop-files.zip)

**Reply by abenz1267** [reply]
Thanks, i'll look into it at some point.... rn i'm in the middle of moving, so it might take a while.

**Reply by victorkamoto** [reply]
Hello @abenz1267 , did you manage to have a look?

**Reply by abenz1267** [reply]
Hey,

sorry, I'm super busy with still moving and work these days. I honestly forgot about this. I'll report back... at some point...

Regards

**Reply by abenz1267** [reply]
Hey, so i finally tested out with all your desktop files.... but i sadly couldn't find bad performance on my system.
